python/ray/serve/_private/proxy.py

```diff
@@ -1039,19 +1039,24 @@ async def send_request_to_replica(
         finally:
             if not proxy_asgi_receive_task.done():
                 proxy_asgi_receive_task.cancel()
-            else:
-                # If the server disconnects, status_code is set above from the
-                # disconnect message. Otherwise the disconnect code comes from
-                # a client message via the receive interface.
-                if (
-                    status is None
-                    and proxy_request.request_type == "websocket"
-                    and proxy_asgi_receive_task.exception() is None
-                ):
+
+            # If the server disconnects, status_code can be set above from the
+            # disconnect message. but it is not guaranteed.
+            # If client disconnects, the disconnect code comes from
+            # a client message via the receive interface, but is is not guaranteed.
+            if status is None and proxy_request.request_type == "websocket":
+                if not proxy_asgi_receive_task.cancelled:
```

Comment:
Should this be `cancelled()`? https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.cancelled

Reply:
good call